### ISSUE TITLE
Timeout: request did not complete within allowed duration

### ISSUE THREAD

**Issue by sunshanpeng** [denverdino/lxcfs-initializer#3]
内核3.10，centos7.5，k8s1.11.3
```bash
[root@master1 ~]# cat /etc/redhat-release
CentOS Linux release 7.5.1804 (Core) 
[root@master1 ~]# uname -a
Linux master1 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
[root@master1 ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T18:02:47Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.3", GitCommit:"a4529464e4629c21224b3d52edfe0ea91b072862", GitTreeState:"clean", BuildDate:"2018-09-09T17:53:03Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```
**使用daemonset部署后，加`annotations:    "initializer.kubernetes.io/lxcfs": "true"`的容器能正常启动，不加`annotations:    "initializer.kubernetes.io/lxcfs": "true"`的容器启动时都报错，信息如下：**
```bash
[root@master1 ~]# kubectl apply -f test.yaml 
service/nginx-service created
Error from server (Timeout): error when creating "test.yaml": Timeout: request did not complete within allowed duration
```
文件内容如下
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  #annotations:
    # "initializer.kubernetes.io/lxcfs": "true"
  name: nginx-deployment
spec:
  replicas: 2
  selector:
    matchLabels:
      app: nginx 
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx 
        image: nginx:1.14 
        imagePullPolicy: Always 
        ports: 
        - name: service-port
          containerPort: 80 
        resources: 
          requests: 
            memory: "250Mi"
            cpu: "0.5"
          limits: 
            memory: "500Mi"
            cpu: "1"
        volumeMounts:
            - name: timezone
              mountPath: /etc/localtime
      restartPolicy: Always 
      volumes:
        - name: timezone
          hostPath:
            path: /usr/share/zoneinfo/Asia/Shanghai

---

apiVersion: v1
kind: Service
metadata:
  name: nginx-service
spec:
  ports:
  - port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: nginx
```
lxcfs-daemonset和lxcfs-initializer也是正常启动的
```bash
[root@master1 ~]# kubectl get pods
NAME                                 READY     STATUS    RESTARTS   AGE
lxcfs-cn76r                          1/1       Running   0          1h
lxcfs-initializer-647c5fb868-7mkfg   1/1       Running   0          1h
lxcfs-v75qt                          1/1       Running   0          1h
lxcfs-x5jdl                          1/1       Running   0          1h
```

**Comment by level077** [comment]
编辑lxcfs-initializer.yaml，添加update权限进去就不会报错，错误信息你可以查看lxcfs-initializer的log
但是在容器里看到的还是宿主机的资源

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: lxcfs-initializer-default
  namespace: default
rules:
- apiGroups: ["*"]
  resources: ["deployments"]
  verbs: ["initialize", "patch", "watch", "list","update"]
```

**Comment by denverdino** [comment]
Thanks, the "update" is required.

Fixed and tested with K8s 1.11
